crates/chain/src/indexer/keychain_txout.rs

```diff
@@ -254,14 +254,14 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     /// Return the script that exists under the given `keychain`'s `index`.
     ///
     /// This calls [`SpkTxOutIndex::spk_at_index`] internally.
-    pub fn spk_at_index(&self, keychain: K, index: u32) -> Option<&Script> {
+    pub fn spk_at_index(&self, keychain: K, index: u32) -> Option<ScriptBuf> {
         self.inner.spk_at_index(&(keychain.clone(), index))
```

Comment:
What happens if we enforce that `K` is also `Copy`.

Ref https://github.com/bitcoindevkit/bdk/blob/37cef1da7834d6d29f91c382ae841baf66764c51/crates/chain/src/indexer/keychain_txout.rs#L137

I see from the example in the docstrings that probably `K` can be trait-bounded to `Copy` and then we don't need all those `.clone()`

Reply:
Limiting K to have to implement `Copy` I think would be too restrictive. For instance we wouldn't be able to use types like `String` as K or as fields of a K.

Reply:
Yes, then `clone()` is fine.